tools/installer/installer.py

```diff
@@ -0,0 +1,940 @@
+        self._is_upgrade = iu
+
+    @is_from_scratch.setter
+    def is_from_scratch(self, ifs):
+        self._is_from_scratch = ifs
+
+    @rewrite_systemd.setter
+    def rewrite_systemd(self, rs):
+        self._rewrite_systemd = rs
+
+    @rewrite_rsyslog.setter
+    def rewrite_rsyslog(self, rr):
+        self._rewrite_rsyslog = rr
+
+    @rewrite_logrotate.setter
+    def rewrite_logrotate(self, rl):
+        self._rewrite_logrotate = rl
+
+    @is_cosmo_needed.setter
+    def is_cosmo_needed(self, icn):
+        self._is_cosmo_needed = icn
+
+    @init_from_snapshot.setter
+    def init_from_snapshot(self, ifs):
+        self._init_from_snapshot = ifs
+
+    @chain.setter
+    def chain(self, chain):
+        self._chain = chain
+
+    @snapshot_url.setter
+    def snapshot_url(self, su):
+        self._snapshot_url = su
+
+    @is_setup_needed.setter
+    def is_setup_needed(self, is_setup_needed):
+        self._is_setup_needed = is_setup_needed
+
+    @moniker.setter
+    def moniker(self, moniker):
+        self._moniker = moniker
+
+    @external_address.setter
+    def external_address(self, external_address):
+        self._external_address = external_address
+
+    @rpc_port.setter
+    def rpc_port(self, rpc_port):
+        self._rpc_port = rpc_port
+
+    @p2p_port.setter
+    def p2p_port(self, p2p_port):
+        self._p2p_port = p2p_port
+
+    @gas_price.setter
+    def gas_price(self, gas_price):
+        self._gas_price = gas_price
+
+    def log(self, msg):
+        if self.verbose:
+            print(f"{PRINT_PREFIX} {msg}")
+
+    def is_already_installed(self) -> bool:
+        return os.path.exists(self.home_dir) and \
+            os.path.exists(self.cheqd_root_dir)
+
+    def is_systemd_config_exists(self) -> bool:
+        return os.path.exists(DEFAULT_COSMOVISOR_SERVICE_FILE_PATH) or \
+            os.path.exists(DEFAULT_STANDALONE_SERVICE_FILE_PATH)
+
+
+    def get_releases(self):
+        req = request.Request("https://api.github.com/repos/cheqd/cheqd-node/releases")
+        req.add_header("Accept", "application/vnd.github.v3+json")
+
+        with request.urlopen(req) as response:
+            r_list = json.loads(response.read().decode("utf-8"))
+            return [Release(r) for r in r_list]
+
+    def get_last_prerelease(self, r_list) -> Release:
+        for r in r_list:
+            if not r.is_prerelease:
+                return r
+
+    def ask_for_version(self):
+        all_releases = self.get_releases()
+        default = self.get_last_prerelease(all_releases)
+        self.log(f"Latest stable cheqd-noded release version is {default}")
+        d_index = all_releases.index(default)
+        last_n_releases = all_releases[d_index:LAST_N_RELEASES]
+        for i, release in enumerate(last_n_releases):
+            print(f"{i + 1}) {release.version}")
+        num = self.ask("Choose list option below to select version of cheqd-node to install [default: 1]: ", 
+            default=1).stdin.strip()
```

Comment:
Why do we need `stdin` here? Cause looks like `str` object doesn't have such method. And linter also said https://github.com/cheqd/cheqd-node/runs/7124655827?check_suite_focus=true